src/rust/engine/process_execution/src/remote_cache_tests.rs

```diff
@@ -183,6 +185,36 @@ async fn skip_cache_on_error() {
   assert_eq!(result.exit_code, 0);
 }
 
+/// With eager_fetch enabled, we should skip the remote cache if any of the process result's
+/// digests are invalid. This will force rerunning the process locally. Otherwise, we should use
+/// the cached result with its non-existent digests.
+#[tokio::test]
+async fn eager_fetch() {
+  async fn run_process(eager_fetch: bool) -> FallibleProcessResultWithPlatform {
+    let (local, store, _local_runner_dir, _stub_cas) = create_local_runner();
+    let (caching, _cache_dir, stub_action_cache) =
+      create_cached_runner(local, store.clone(), eager_fetch);
+
+    // TODO: how to insert a cache entry without actually running it? I think
+    // `stub_action_cache.action_map.clone().lock().insert()`? I need the Process's digest
+    // to be the same as the below `process` var, but the ProcessResult digests should be fake.
```

Comment:
@stuhood @tdyas any thoughts? 

If you have a moment, it might be more effective to pair on this last step of getting the test working. (With Slack down, we can email to set this up.)

Reply:
It looks like you can use https://github.com/pantsbuild/pants/blob/768b3a3aad4200d6372732ad0447d6558b31d04a/src/rust/engine/process_execution/src/remote_cache.rs#L297-L308 (possibly into a throwaway `Store` to ensure that the digests don't end up anywhere readable)?

Reply:
Sorry... perhaps you need both `ensure_action_stored_locally` and `make_action_result` to construct the key and the value, respectively.